### PR TITLE
Feat: allow empty policy name

### PR DESCRIPTION
**Pull request by Somefive**
### Description of your changes

Support using anonymous policy, like
```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: test
spec:
  components:
    - type: webservice
      name: test
      properties:
        image: nginx
  policies:
    - type: topology
      properties:
        clusters: ["cluster-worker"]
    - type: override
      properties:
        components:
          - properties:
              image: nginx:1.21
```

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/appfile/parser.go [hunk 1]
```diff
@@ -112,6 +112,12 @@ func (p *Parser) GenerateAppFileFromApp(ctx context.Context, app *v1beta1.Applic
 	ns := app.Namespace
 	appName := app.Name
 
+	for idx := range app.Spec.Policies {
+		if app.Spec.Policies[idx].Name == "" {
+			app.Spec.Policies[idx].Name = fmt.Sprintf("%s:auto-gen:%d", app.Spec.Policies[idx].Type, idx)
```
Comment (wonderflow) [inline pkg/appfile/parser.go:117]: Does K8s accept the name contains the `:`?

Comment (Somefive) [reply pkg/appfile/parser.go:117]: This is a internal name, not stored in etcd. FYI, it supports ":".

Comment (wonderflow) [reply pkg/appfile/parser.go:117]: It can be used in `context.name`?

**Review by chivalryq — APPROVED** [review]
LGTM

File: pkg/appfile/parser.go (hunk 1 above)
Comment (Somefive) [reply pkg/appfile/parser.go:117]: No. There is no need for `context.name`. `context.name` currently refers to either `workflowstep` or `component`.

**Pull request merged**